src/lib.rs

```diff
@@ -145,10 +149,62 @@ mod tests {
     // #[serial]
     // fn test_display_address_with_path_taproot() {}
 
-    // TODO: Create PSBT with scratch using given Hardware Wallet
-    // #[test]
-    // #[serial]
-    // fn test_sign_tx() {}
+    #[test]
+    #[serial]
+    fn test_sign_tx() {
+        let devices = HWIClient::enumerate().unwrap();
+        let device = devices.first().unwrap();
+        let client = HWIClient::get_client(device, true, types::HWIChain::Test).unwrap();
+        let derivation_path = DerivationPath::from_str("m/44'/1'/0'/0/0").unwrap();
+
+        let address = client
+            .display_address_with_path(&derivation_path, types::HWIAddressType::Legacy)
+            .unwrap();
+
+        let pk = client.get_xpub(&derivation_path, true).unwrap();
+        let mut hd_keypaths: BTreeMap<secp256k1::PublicKey, KeySource> = Default::default();
+        hd_keypaths.insert(pk.public_key, (device.fingerprint, derivation_path));
```

Comment:
You should put the master xpub fingerprint in the map, which in this case it's the same as the device fingerprint. It's ok to leave it like that, but please add a small comment explaining